### PR TITLE
Dreamview:Open space planner dreamview on 5.5

### PR DESCRIPTION
**Pull request by lj0919**

### CODE REVIEW

File: modules/dreamview/frontend/src/renderer/index.js [hunk 1]
```diff
@@ -480,7 +496,10 @@ class Renderer {
         if (removeOldMap) {
             this.map.removeAllElements(this.scene);
         }
-        this.map.appendMapData(newData, this.coordinates, this.scene);
+        const adjustOrder=this.map.appendMapData(newData, this.coordinates, this.scene);
```
Comment (changsh726) [inline modules/dreamview/frontend/src/renderer/index.js:499]: spaces between `=`

File: modules/dreamview/frontend/src/renderer/index.js [hunk 2]
```diff
@@ -480,7 +496,10 @@ class Renderer {
...
+        if (newData['parkingSpace'] && !_.isEmpty(adjustOrder)) {
+            this.routingEditor.setParkingSpaceInfo(newData['parkingSpace'],adjustOrder,this.coordinates);
```
Comment (changsh726) [inline modules/dreamview/frontend/src/renderer/index.js:501]: space after `,`

File: modules/dreamview/backend/simulation_world/simulation_world_updater.cc [hunk 3]
```diff
@@ -477,6 +498,43 @@ bool SimulationWorldUpdater::ConstructRoutingRequest(
...
+  parking_routing_task->clear_routing_point();
+  // set parking Space
+  if (!ContainsKey(json, "parkingSpace")) {
+    AERROR << "Failed to prepare a parking routing task: "
+           << "parking space not found.";
+    return false;
+  }
+  auto *requested_parking_space = parking_routing_task->mutable_parking_space();
+  if (!JsonStringToMessage(json["parkingSpace"].dump(), requested_parking_space)
+  .ok()) {
```
Comment (changsh726) [inline modules/dreamview/backend/simulation_world/simulation_world_updater.cc:512]: The format here looks a little odd

File: modules/dreamview/backend/simulation_world/simulation_world_updater.cc [hunk 4]
```diff
@@ -477,6 +498,43 @@ bool SimulationWorldUpdater::ConstructRoutingRequest(
...
+    AERROR << "Failed to prepare a parking routing task: invalid parking space."
+             << json["parkingSpace"].dump();
+             return false;
```
Comment (changsh726) [inline modules/dreamview/backend/simulation_world/simulation_world_updater.cc:515]: and these two lines

File: modules/dreamview/backend/simulation_world/simulation_world_updater.cc [hunk 5]
```diff
@@ -477,6 +498,43 @@ bool SimulationWorldUpdater::ConstructRoutingRequest(
...
+        AERROR << "Failed to prepare a parking routing task: invalid point.";
+        return false;
+      }
+      if (!map_service_->ConstructLaneWayPoint(point["x"], point["y"],
+                                               routing_point->Add())) {
+        AERROR << "Failed to construct a LaneWayPoint, skipping.";
+        routing_point->RemoveLast();
+      }
+    }
+    return true;
```
Comment (changsh726) [inline modules/dreamview/backend/simulation_world/simulation_world_updater.cc:535]: wrong indent from line 522 to 535

File: modules/dreamview/frontend/src/renderer/index.js [hunk 6]
```diff
@@ -335,19 +335,30 @@ class Renderer {
     }
 
     removeInvalidRoutingPoint(pointId, error) {
-        this.routingEditor.removeInvalidRoutingPoint(pointId, error, this.scene);
+        const index = this.routingEditor.removeInvalidRoutingPoint(pointId, error, this.scene);
+        if ( index!== -1) {
```
Comment (changsh726) [inline modules/dreamview/frontend/src/renderer/index.js:339]: if (index !== -1)

File: modules/dreamview/backend/simulation_world/simulation_world_updater.cc (hunk 5 above)
Comment (lj0919) [reply modules/dreamview/backend/simulation_world/simulation_world_updater.cc:535]: thx

File: modules/dreamview/frontend/src/renderer/routing_editor.js [hunk 7]
```diff
@@ -48,56 +51,100 @@ export default class RoutingEditor {
...
       object.material.dispose();
     }
+    return index;
   }
 
   sendRoutingRequest(carOffsetPosition, carHeading, coordinates, routingPoints) {
+    // parking routing request vs common routing request
     if (this.routePoints.length === 0 && routingPoints.length === 0) {
       alert("Please provide at least an end point.");
         return false;
```
Comment (changsh726) [inline modules/dreamview/frontend/src/renderer/routing_editor.js:128]: wrong indent

File: modules/dreamview/frontend/src/utils/draw.js [hunk 8]
```diff
@@ -188,3 +188,16 @@ export function disposeMesh(mesh) {
     mesh.geometry.dispose();
     mesh.material.dispose();
 }
+
+export function changeMaterial(mesh, color=0xff0000, linewidth=2, transparent=false, opacity=1) {
+    if (!mesh) {
+        return;
+    }
+    mesh.material.dispose();
+    mesh.material=new THREE.LineBasicMaterial({
```
Comment (changsh726) [inline modules/dreamview/frontend/src/utils/draw.js:197]: Add spaces to both sides of `=`

File: modules/task_manager/proto/BUILD [hunk 9]
```diff
@@ -15,6 +15,7 @@ proto_library(
     deps = [
         "//modules/common/proto:header_proto_lib",
         "//modules/routing/proto:routing_proto_lib",
+        "//modules/map/proto:map_proto_lib",
```
Comment (changsh726) [inline modules/task_manager/proto/BUILD:18]: In alphabetical order

File: modules/dreamview/frontend/src/renderer/map.js [hunk 10]
```diff
@@ -493,5 +519,169 @@ export default class Map {
...
+    }
+
+    // find parking space based which lane
+    getLaneBasedOverlapId(overlapId, coordinates) {
+        // Default based one lane if overlapId.length>=2 only choose the first
+        let result = [];
+        if (overlapId.length !== 0) {
+            const laneIndex = _.findIndex(this.data['lane'], item => {
+                return _.findIndex(item.overlapId, { 'id': overlapId[0].id }) !== -1;
+
```
Comment (changsh726) [inline modules/dreamview/frontend/src/renderer/map.js:536]: remove the blank line

File: modules/dreamview/frontend/src/renderer/map.js [hunk 11]
```diff
@@ -493,5 +519,169 @@ export default class Map {
...
+                        orderIndex = pointWithDirectionVector(border[1], mid03, mid12) ? 1 : 2;
+                    } else {
+                        //12->03
+                        orderIndex = pointWithDirectionVector(border[0], mid12, mid03) ? 0 : 3;
+                    }
+                }
+            }
+            result = order2[orderIndex];
+        }
+        console.log(result);
```
Comment (changsh726) [inline modules/dreamview/frontend/src/renderer/map.js:684]: Remove this after debugging.

Comment (lj0919) [reply modules/dreamview/frontend/src/renderer/map.js:684]: ok

File: modules/dreamview/frontend/src/renderer/map.js (hunk 10 above)
Comment (lj0919) [reply modules/dreamview/frontend/src/renderer/map.js:536]: thx

File: modules/dreamview/frontend/src/renderer/index.js (hunk 6 above)
Comment (lj0919) [reply modules/dreamview/frontend/src/renderer/index.js:339]: thx

**Pull request closed without merging**